src/viper/trans.ml

```diff
@@ -515,7 +515,7 @@ and stmt ctxt (s : M.exp) : seqn =
   | M.WhileE(e, s1) ->
      let (invs, s1') = extract_loop_invariants s1 in
      let invs' = List.map (fun inv -> exp ctxt inv) invs in
-     let invs'' = invs' @ local_access_preds ctxt in
+     let invs'' = local_access_preds ctxt @ invs' in
```

Comment:
Is it necessary?

Reply:
Yeah, we want to request access _before_ writing down user predicates

Reply:
oh, right, I didn't notice. Thx